html2asketch/helpers/convertAngleToFromAndTo.js

```diff
@@ -1,39 +1,52 @@
 // Keep this pure for easy testing in the future.
 export default function convertAngleToFromAndTo(angle) {
-  // default 180deg
-  const from = {x: 0.5, y: 0};
-  const to = {x: 0.5, y: 1};
-
-  // Learn math or find someone smarter to figure this out correctly
-  switch (angle) {
-    case 'to top':
-    case '360deg':
-    case '0deg':
-      from.y = 1;
-      to.y = 0;
-      break;
-    case 'to right':
-    case '90deg':
-      from.x = 0;
-      from.y = 0.5;
-      to.x = 1;
-      to.y = 0.5;
-      break;
-    case 'to left':
-    case '270deg':
-      from.x = 1;
-      from.y = 0.5;
-      to.x = 0;
-      to.y = 0.5;
-      break;
-    case 'to bottom':
-    case '180deg':
-    default:
-      break;
+  function convertAngleInDegreesToFromAndTo(degrees) {
+    // we begin "from" (0,0) and have a circle (returned as "to") with starting from point at (1,0) for 0deg (to right)
+    // default 0deg
+    const from = {x: 0, y: 0};
+    const to = {x: 1, y: 0};
+
+    if (degrees === 0) {
+      return {from, to};
+    }
+
+    // sin() and cos() needs radians to work with
+    const angleInRadians = degrees * (Math.PI / 180);
+
+    // calculate circle's (x,y) wich is sin nad cos of given radians
+    // we support precision to make things human readable
+    const precision = 2;
+    let x = Number.parseFloat(Math.cos(angleInRadians).toFixed(precision));
+    let y = Number.parseFloat(Math.sin(angleInRadians).toFixed(precision));
+
+    // convert -0 to +0, sin() and cos() may return -0 wich can lead to some miss-calcualtions
+    x += 0;
+    y += 0;
+
+    return {
+      from,
+      to: {x, y},
+    };
+  }
+
+  // we need angle to be a float number
+  const degrees = Number.parseFloat(angle);
```

Comment:
I think to have this generic version working we need to parse angle first. We may get: `[0-360]deg` or `[0-1]turn`.